### ISSUE TITLE
fastapi >= 0.111.0 dependency

### ISSUE THREAD

**Issue by arthurbrenno** [run-llama/llama_deploy#109]
I'm having troubles with llama-agents requiring fastapi >= 0.111.0. Currently, the latest fastapi version is 0.111.0, this would harm other systems that want to use llama agents with lower versions of fastapi. Do you use the fastapi cli tool? . I'm not building a library, so I'm just specifying the exact versions of the packages I want. Can I somehow still use llama-agents and a lower version of fastapi?

```
pip compile requirements.in -o requirements.txt
  × No solution found when resolving dependencies:
  ╰─▶ Because llama-agents==0.0.3 depends on fastapi>=0.111.0 and you require fastapi==0.110.0, we can conclude that your requirements and
      llama-agents==0.0.3 are incompatible.
      And because you require llama-agents==0.0.3, we can conclude that the requirements are unsatisfiable.
make: *** [Makefile:14: compile] Error 1
```

**Comment by logan-markewich** [comment]
We can definitely lower. I just did `poetry add fastapi` without thinking about the version too much 😅

**Comment by logan-markewich** [comment]
Ok, so due to security issues, the lowest I can go is v0.109 

But, should be merged + released with v0.0.7
